silta-cluster/templates/nfs-server.yaml

```diff
@@ -40,25 +40,74 @@ spec:
           tcpSocket:
             port: 111
         resources:
-          {{- index (index .Values "nfs-server") "resources" | toYaml | nindent 10 }}
+          {{- $nfsserver.resources | toYaml | nindent 10 }}
         volumeMounts:
+        {{- if $nfsserver.persistence.rclone.enabled }}
+        - name: exports
+          mountPath: /exports
+          mountPropagation: Bidirectional
+        {{- else }}
         - name: {{ .Release.Name }}-nfs-server-data
           mountPath: /exports
+        {{- end }}
+      {{- if $nfsserver.persistence.rclone.enabled }}
+      - name: rclone
+        image: {{ $nfsserver.persistence.rclone.image }}
+        imagePullPolicy: {{ $nfsserver.persistence.rclone.imagePullPolicy }}
+        # TODO: cache modes and other options + expose flags in values.yaml
+        command: [
+          "/usr/local/bin/rclone", 
+          "mount", 
+          "backend:{{ $nfsserver.persistence.rclone.remotePath }}", 
+          "/exports", 
+          "--allow-non-empty", 
+          "--allow-other",
+          {{- range $key, $value := $nfsserver.persistence.rclone.params }}
+          "--{{ $key }}={{ $value }}",
+          {{- end }}
+        ]
+        securityContext:
+          # Required by fuse mount
+          privileged: true
+          capabilities:
+            add:
+              - SYS_ADMIN
+        resources:
+          {{- $nfsserver.persistence.rclone.resources | toYaml | nindent 10 }}
+        volumeMounts:
+          - name: rclone-config
+            mountPath: /config/rclone/rclone.conf
+            subPath: rclone.conf
+          - name: exports
+            mountPath: /exports
+            mountPropagation: Bidirectional
+      {{- end }}
+      volumes:
+      - name: rclone-config
+        configMap:
+          name: {{ .Release.Name }}-nfs-server-rclone-config
```

Comment:
The configmap is created only if rclone is enabled, so this volume item should probably be wrapped in an if-statement, right? I guess you should move the if from line 89 to cover the whole volumes object.

Reply:
change added